src/libraries/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -11,6 +11,231 @@ namespace System.Text.Json.Serialization.Tests
 {
     public static class PropertyVisibilityTests
```

Comment:
It's a naming used by EF Core for long test names, but am okay with changing this if you want, but first compare names.

**Before**: `Serialize_base_public_property_on_conflict_with_derived_private`
**After**: `SerializeBasePublicPropertyOnConflictWithDerivedPrivate`

The first one is more readable while the second looks like encoded data and is hard to read. So are you sure that pascal case is better? I can rename other tests and give them meaningful names instead of current (:

Reply:
I do underscore-delimited pascal case chunks in such cases, e.g. `Serialize_BasePublicProperty_OnConflictWithDerivedPrivate`. The **before** pattern isn't used anywhere else in this codebase.